extensions/wikia/Insights/specials/UnconvertedInfoboxesPage.class.php

```diff
@@ -95,27 +95,30 @@ public function recache( $limit = false, $ignoreErrors = true ) {
 	 * @return bool|mixed
 	 */
 	public function reallyDoQuery( $limit = false, $offset = false ) {
-		$dbr = wfGetDB( DB_SLAVE, [ $this->getName(), __METHOD__, 'vslow' ] );
-
-		$nonportableTemplates = ( new WikiaSQL() )
-			->SELECT( 'page_title' )->AS_( 'title' )
-			->FROM( 'page' )
-			->WHERE( 'page_namespace' )->EQUAL_TO( NS_TEMPLATE )
-				->AND_( 'page_is_redirect' )->EQUAL_TO( 0 )
-			->runLoop( $dbr, function( &$nonportableTemplates, $row ) {
-				$title = Title::newFromText( $row->title, NS_TEMPLATE );
-				$contentText = ( new WikiPage( $title ) )->getText();
-				if ( $title !== null && PortableInfoboxClassification::isTitleWithNonportableInfobox( $title->getText(), $contentText ) ) {
+		global $wgCityId;
+
+		$tcs = new UserTemplateClassificationService();
+		$recognizedTemplates = $tcs->getTemplatesOnWiki( $wgCityId );
+
+		$nonportableInfoboxes = [];
+
+		foreach ( $recognizedTemplates as $templateId => $type ) {
+			if ( $tcs->isInfoboxType( $type ) ) {
+				$title = Title::newFromID( $templateId );
+				if ( $title instanceof Title
+					&& empty( PortableInfoboxDataService::newFromTitle( $title )->getData() )
```

Comment:
We have just last use of PortableInfoboxClassification::isTitleWithNonportableInfobox in our code which is in TemplateDraftHooks::onEditPageLayoutShowIntro. Can we unify this as well and remove isTitleWithNonportableInfobox? I think it's worth to unify it as well because TemplateDtaft is part of Insights loop.